feat(issues): Add context tooltips for issue header status

JoshFerge opened the pull request:
For a new Sentry user, the issue header is an overwhelming amount of information, and it's not clear what the information is.

We add tooltips for all issue statuses and provide a link to learn more in our docs.


refs https://linear.app/getsentry/issue/RTC-854/issue-details-add-explanatory-mouseovers-to-titles


<img width="221" alt="Screenshot 2025-05-21 at 7 11 48 PM" src="https://github.com/user-attachments/assets/193735d0-74ec-4066-8f29-c864e2db75da" />

JoshFerge committed:
feat(issues): add context tooltips for issue hader

scttcper commented on static/app/views/issueDetails/streamline/header/header.tsx:
```diff
@@ -178,15 +186,34 @@ export default function StreamlinedGroupHeader({
             {statusProps?.status ? (
               <Fragment>
                 <Tooltip title={statusProps?.tooltip}>
```
double tooltips

JoshFerge replied:
ah i did not notice this. when does status show a from this?

scttcper replied:
looks like here https://github.com/getsentry/sentry/blob/4525724c64472bec454b994a8f1b978940d30246/static/app/components/group/inboxBadges/statusBadge.tsx#L51-L57

scttcper replied:
you could add your tooltips there or fallback to a default tooltip or something

scttcper commented on static/app/views/issueDetails/streamline/header/header.tsx:
```diff
@@ -178,15 +186,34 @@ export default function StreamlinedGroupHeader({
             {subtitle && (
               <Fragment>
                 <Divider />
                 <Subtitle title={subtitle} isHoverable showOnlyOnOverflow delay={1000}>
```
subtitle here is also a tooltip

scttcper commented on static/app/views/issueDetails/streamline/header/header.tsx:
```diff
@@ -156,12 +157,19 @@ export default function StreamlinedGroupHeader({
-            message={secondaryTitle}
-            type={group.type}
-          />
+          <Tooltip
+            title={t(
+              'The main message summarizing the event. Typically the error message or log string that triggered this issue.'
+            )}
+            isHoverable
+          >
```
use skipWrapper where possible which basically just doesn't render an extra span. If you're wrapping a div or an existing span you don't need another span.
```suggestion
          <Tooltip
            title={t(
              'The main message summarizing the event. Typically the error message or log string that triggered this issue.'
            )}
            isHoverable
            skipWrapper
          >
```

JoshFerge committed:
Merge branch 'master' into jferg/learn-more

JoshFerge committed:
add more status tooltip messages, and remove other tooltips

JoshFerge commented:
I've changed this PR to focus on the status tooltip, as that one is easier than the other 3. I've also added info texts describing the other statuses.

JoshFerge commented on static/app/components/group/inboxBadges/statusBadge.tsx:
```diff
@@ -18,30 +18,37 @@ export function getBadgeProperties(
       return {
         tagType: 'error',
         status: t('Escalating'),
+        tooltip: t('This issue has exceeded its forecasted event volume.'),
```
this one could be improved i think.

mrduncan replied:
I don't particularly like my phrasing here but maybe something along the lines of "This issue is occurring significantly more often it used to."

https://docs.sentry.io/product/issues/states-triage/escalating-issues/ says "Sentry defines an issue as Escalating when the number of events is significantly higher than the previous week." but there is also the caveat for issues <7d old

JoshFerge replied:
i like this a lot better!

JoshFerge committed:
update copy for escalating

JoshFerge committed:
fix test
